python_modules/dagster/dagster/_core/storage/schedules/schema.py

```diff
@@ -45,6 +47,24 @@
     db.Column("update_timestamp", db.DateTime, server_default=get_current_timestamp()),
 )
 
+AssetPolicyEvaluationsTable = db.Table(
+    "asset_policy_evaulations",
+    ScheduleStorageSqlMetadata,
+    db.Column(
+        "id",
+        db.BigInteger().with_variant(sqlite.INTEGER(), "sqlite"),
+        primary_key=True,
+        autoincrement=True,
+    ),
+    db.Column("evaluation_id", db.BigInteger().with_variant(sqlite.INTEGER(), "sqlite")),
+    db.Column("asset_key", db.Text, index=True),
+    db.Column("evaluation_body", db.Text),
+    db.Column("policy_type", db.String(63)),
```

Comment:
do we need policy_type?

Reply:
ah I see why we might want to keep track of what the policy was at the time of given evaluation. however, I think to do this completely correctly, we'd need to serialize the whole AutoMaterializePolicy NamedTuple.

this is probably a good idea though -- means that users won't be confused if they update their policy and look at a historical tick that would have made a different decision with the new policy

Reply:
That's actually not what I was thinking of, but you raise a good point. I wonder if we'd need to serialize the policy each time though- we could probably use a snapshot id. I think we could add that in the future though

Reply:
My thought was this would store AUTO_MATERIALIZE for each row to future proof the table in the event that we added asset expectations etc. to it in the future. Per discussion here https://github.com/dagster-io/dagster/pull/14185#discussion_r1189245250 maybe it'd be `evaluation_type`?

Reply:
What precisely is the meaning of "Evaluation" in this Ontology? CC @OwenKephart

Reply:
In Josh's designs the table is labeled evaluations, so I think of an evaluation as equal to one row in the per asset table. Once instance of evaluating whether an asset should materialize or not. Asset daemon evaluations does sound more like there'd be 1 per loop though